### ISSUE TITLE
Problem resolving type parameters of XmlAdapter

### ISSUE THREAD

**Issue by rcpoison** [stoicflame/enunciate#156]
Hi ^^
Using v2.0.0-RC.1 all enunciate modules fail to build.
This seems to come down to the fact that I've managed to confuse enunciate with my XmlAdapters:
```
public abstract class AbstractEntityAdapter<K, T extends EntityInterface<K>> extends XmlAdapter<K, T>

public class BeanBAdapter extends AbstractEntityAdapter<Integer, BeanB>

```
enunciate will resolve the type to K instead of Integer.

This is probably related to: https://github.com/stoicflame/enunciate/issues/147, but I can't tell for sure.

I've created a small test project at: https://github.com/rcpoison/enunciate2test

**Comment by rcpoison** [comment]
Sorry, I meant all the client modules and forgot the error:
```
/home/poison/files/dev/enunciate2test/target/enunciate/java-xml-client/src/com/ignorelist/kassandra/enunciate2test/BeanA.java:24: error: cannot find symbol
  private K _b;
          ^
  symbol:   class K
  location: class BeanA
/home/poison/files/dev/enunciate2test/target/enunciate/java-xml-client/src/com/ignorelist/kassandra/enunciate2test/BeanA.java:51: error: cannot find symbol
  public K getB() {
         ^
  symbol:   class K
  location: class BeanA
/home/poison/files/dev/enunciate2test/target/enunciate/java-xml-client/src/com/ignorelist/kassandra/enunciate2test/BeanA.java:58: error: cannot find symbol
  public void setB(K _b) {
                   ^
  symbol:   class K
  location: class BeanA
```

**Comment by stoicflame** [comment]
Fixed at 49c9cc1. I've deployed 2.0.0-SNAPSHOT; you're welcome to try it out if you want to use the [OSSRH snapshot repository](https://oss.sonatype.org/content/repositories/snapshots/).

**Comment by rcpoison** [comment]
That was fast! Working, thanks!
